enhancements/improve-kubeletconfig-default-configuration-check-enhancement.md

```diff
@@ -0,0 +1,285 @@
+---
+title: kubeletconfig-default-configuration-check
+authors:
+  - Vincent056
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - TBD
+approvers:
+  - TBD
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - TBD
+creation-date: 2022-07-15
+last-updated: 2022-07-15
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+---
+
+# KubeletConfig Default Configuration Check
+
+## Summary
+
+This will allow us to improve `KubeletConfig` checks by adding the ability to
+check the default configuration of the `KubeletConfig`. We're aggregating the
+configs into a single place to make it easier to write content against the
+`KubeletConfig`.
+
+## Motivation
+
+Currently, all the `KubeletConfig` checks fails unless they are explicitly
+set by the user. And to set the `KubeletConfig`, it will trigger a node reboot. If
+we enable Compliance Operator to check the default configuration of the `KubeletConfig`,
+a reboot might be avoided when the default configuration meets the compliance requirements.
+
+
+### Goals
+
+* Compliance Operator will be able to check the default configuration 
+of the `KubeletConfig`.
+
+* All the KubeletConfig rules needs to be updated to work with this enhancement.
```

Comment:
Right now, if we write a rule that checks a configuration option that doesn't exist in the configuration file because the deployment is using the default, the rule will FAIL.

I had the impression that rules that failed before because a missing configuration would just start reporting as PASS or FAIL based on the real value of the configuration option. I think I'm failing to understand why the rules need to change to use this new feature.

Unless that particular problem isn't being addressed by this feature?

Reply:
Because the current rules check `/etc/kubernetes/kubelet.conf`. Yes, there is a `KubeletConfig` CR which is what the remediations modify, but the end result is written to `kubelet.conf` on the node. Is tagged with `ocp4-node` and while evaluating the rule, oscap mounts the host fs and scans the host fs as a chroot.

In comparison, the new rules read the proxy object which is an etcd resource. I think most of the rules contents could be reused, but not the rules as a whole, unfortunately.

Reply:
Ah - so we need to change each rule so they check the new kubelet configuration location instead of `/etc/kubernetes/kubelet.conf`.

Reply:
> What's unclear to me is if we care about using old content with new CO - I think the answer is no, though. Customers typically tend to use both as one unit.

We currently don't support mixing and matching CO with different content, do we? I'm not aware of any statements that claim support for that.

Reply:
The new content will only work with newer CO version. I will add that to the doc.